### DISCUSSION TITLE
[REQ] Bot logs over hosted website possible?

### DISCUSSION THREAD

**Discussion by DineshValor** [masterking32/MasterHamsterKombatBot#313] [Q&A] [answered]
I was thinking, is it possible to show logs on website (live) ?

Like [here](https://hamstervalor.onrender.com/) hosted the bot and made a web page, so easily visiting there to check bot running logs!

**Comment by qawechi** [comment]
Yup it is possible. There are many free servers u can use. I have a server and i run the bot there. It is 24/7 running without stopping.

**Reply by MARSWALLET** [reply]
Please what server can I use to host it

**Reply by DineshValor** [reply]
> Yup it is possible. There are many free servers u can use. I have a server and i run the bot there. It is 24/7 running without stopping.

Bro I'm talking about how to get bot logs on web page! My bot already hosted on server.

**Reply by qawechi** [reply, marked as answer]
1. **Create a Flask Application**:
   Flask is a lightweight web framework for Python that can help you create a web server to display your script's output.

2. **Run Your Script and Capture Output**:
   Use Python's `subprocess` module to run your script and capture its output.

3. **Display the Output on a Webpage**:
   Use Flask to serve the captured output on a public webpage.

Here's an example:

1. **Install Flask**:
   ```bash
   pip install Flask
   ```

2. **Create a Flask Application**:
   ```python
   from flask import Flask, render_template_string
   import subprocess

   app = Flask(__name__)

   @app.route('/')
   def index():
       # Run the script and capture the output
       result = subprocess.run(['python3', 'main.py'], capture_output=True, text=True)
       output = result.stdout
       return render_template_string('<pre>{{ output }}</pre>', output=output)

   if __name__ == '__main__':
       app.run(host='0.0.0.0', port=5000)
   ```

3. **Run the Flask Application**:
   ```bash
   python app.py
   ```

4. **Access the Webpage**:
   Open your browser and go to `http://<your-server-ip>:5000` to see the output of your script.

You can deploy this Flask application on your AWS EC2 instance and make sure the necessary ports are open in your security group settings to allow public access.

**Reply by DineshValor** [reply]
> 1. **Create a Flask Application**:
>    Flask is a lightweight web framework for Python that can help you create a web server to display your script's output.
> 2. **Run Your Script and Capture Output**:
>    Use Python's `subprocess` module to run your script and capture its output.
> 3. **Display the Output on a Webpage**:
>    Use Flask to serve the captured output on a public webpage.
> 
> Here's an example:
> 
> 1. **Install Flask**:
>    ```shell
>    pip install Flask
>    ```
> 2. **Create a Flask Application**:
>    ```python
>    from flask import Flask, render_template_string
>    import subprocess
>    
>    app = Flask(__name__)
>    
>    @app.route('/')
>    def index():
>        # Run the script and capture the output
>        result = subprocess.run(['python3', 'main.py'], capture_output=True, text=True)
>        output = result.stdout
>        return render_template_string('<pre>{{ output }}</pre>', output=output)
>    
>    if __name__ == '__main__':
>        app.run(host='0.0.0.0', port=5000)
>    ```
> 3. **Run the Flask Application**:
>    ```shell
>    python app.py
>    ```
> 4. **Access the Webpage**:
>    Open your browser and go to `http://<your-server-ip>:5000` to see the output of your script.
> 
> You can deploy this Flask application on your AWS EC2 instance and make sure the necessary ports are open in your security group settings to allow public access.

Very helpful, I'm definitely gonna try it, thanks a lot buddy

**Comment by copycatcodex** [comment]
how about if there are updates with the bot?

**Reply by tboy1337** [reply]
Use my updater https://github.com/masterking32/MasterHamsterKombatBot/discussions/290 xD

**Comment by tboy1337** [comment]
> I was thinking, is it possible to show logs on website (live) ?
> 
> Like [here](https://hamstervalor.onrender.com/) hosted the bot and made a web page, so easily visiting there to check bot running logs!

I have made local logging code, how you get that from your machine onto your website is up to you, the admin @masterking32 doesnt want local logging though so be quick before he closes your discussion xD

Information about local logging is in https://github.com/masterking32/MasterHamsterKombatBot/discussions/301

**Reply by DineshValor** [reply]
> > I was thinking, is it possible to show logs on website (live) ?
> > Like [here](https://hamstervalor.onrender.com/) hosted the bot and made a web page, so easily visiting there to check bot running logs!
> 
> I have made local logging code, how you get that from your machine onto your website is up to you, the admin @masterking32 doesnt want local logging though so be quick before he closes your discussion xD
> 
> Information about local logging is in #301

Got it 👍
